exercises/nucleotide-count/example.rs

```diff
@@ -1,13 +1,26 @@
 use std::collections::HashMap;
+use std::collections::hash_map::Entry;
 
-pub fn count(nucleotide: char, input: &str) -> usize {
-    input.chars().filter(|&c| c == nucleotide).count()
+static VALID_NUCLEOTIDES: &'static str = "ACGT";
+
+pub fn count(nucleotide: char, input: &str) -> Result<usize, char> {
+    let valid = |x: char| { VALID_NUCLEOTIDES.contains(x) };
+    if valid(nucleotide) && input.chars().all(valid) {
+        Ok(input.chars().filter(|&c| c == nucleotide).count())
+    } else {
+        Err(nucleotide)
```

Comment:
for `count('A', "AX")` this returns `A` instead of `X`, but we don't check it in the tests so I'm not motivated.